FactoryPattern_API/Airways.cs

```diff
@@ -6,26 +6,55 @@
 
 namespace FactoryPattern_API
 {
-    class Airways :  IProduct
+    class Airways :  Products
     {
         FileDatabase file = new FileDatabase();
         public string Name { get; set; } 
         public bool isBooked { get; set; }
-        public void Book()
+        public int Price { get; set; }
+        string NameOfProduct { get; set; }
+        int PriceOfProduct { get; set; }
+
+        public void GetObjectOfAir(Airways product)
+        {
+            NameOfProduct = product.Name;
+            PriceOfProduct = product.Price;
+        }
+
+        public override void Book(int typeOfDB)
         {
             LogManager.Instance.WriteLog("---Inside Airways/Book---");
-            file.EnterBookingDetails("Booking Done Successfully --- Airways");
+            if (typeOfDB == 1)
```

Comment:
incorrect way of calling which database to use, if tomorrow you introduce another db system you will need to use another if case, something which could have been solved if you would have passed Irepository into this method